From 02f5925b06a9863e8bcadf8efd8624e415fed78b Mon Sep 17 00:00:00 2001
From: Sven Eckelmann <seckelmann@datto.com>
Date: Mon, 12 Aug 2019 11:13:04 +0200
Subject: [PATCH] ath11k: fix multicast rate parameter size

The ATH11K_HW_RATE_CODE() macro uses at least 28 bit to encode the
parameters. An u8 cannot store all these bits.

Fixes: c885aaf227ec ("ath11k: support for multicast rate control")
Signed-off-by: Sven Eckelmann <seckelmann@datto.com>
---
 drivers/net/wireless/ath/ath11k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -1703,7 +1703,7 @@ static void ath11k_mac_op_bss_info_chang
 	u16 bitrate;
 	int ret = 0;
 	u8 rateidx;
-	u8 rate;
+	u32 rate;
 
 	mutex_lock(&ar->conf_mutex);
 
